profiles/views.py

```diff
@@ -0,0 +1,45 @@
+"""Views for courses"""
+from django.shortcuts import get_object_or_404
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.mixins import (
+    RetrieveModelMixin,
+    UpdateModelMixin,
+)
+from rest_framework.viewsets import GenericViewSet
+
+from profiles.models import Profile
+from profiles.serializers import (
+    ProfileSerializer,
+    ProfileLimitedSerializer,
+    ProfilePrivateSerializer,
+)
+from profiles.permissions import CanEditIfOwner
+
+
+class ProfileViewSet(RetrieveModelMixin, UpdateModelMixin, GenericViewSet):
+    """API for the Program collection"""
+    serializer_class_owner = ProfileSerializer
+    serializer_class_limited = ProfileLimitedSerializer
+    serializer_class_private = ProfilePrivateSerializer
+    permission_classes = (IsAuthenticated, CanEditIfOwner)
+    lookup_field = 'user'
+
+    def get_serializer_class(self):
+        """
+        Different parts of a user profile are visible in different conditions
+        """
+        profile = self.get_object()
+        if self.request.user == profile.user:
+            return self.serializer_class_owner
+        elif profile.account_privacy == Profile.PRIVATE:
+            return self.serializer_class_private
+        else:
+            return self.serializer_class_limited
+
+    def get_object(self):
+        """
+        Override get_object to use username
+        """
+        obj = get_object_or_404(Profile, user__username=self.kwargs['user'])
+        self.check_object_permissions(self.request, obj)
```

Comment:
https://github.com/tomchristie/django-rest-framework/blob/master/rest_framework/generics.py#L76

This is the default implementation of get_object. It looks like if we set `lookup_field` to be `user__username` and `lookup_url_kwarg` to `user`, we don't have to write this method at all (and can skip the test mentioned below).